core/src/main/java/com/google/cloud/sql/core/CloudSqlInstance.java

```diff
@@ -56,18 +56,24 @@ class CloudSqlInstance {
   private final CloudSqlInstanceName instanceName;
   private final ListenableFuture<KeyPair> keyPair;
   private final Object instanceDataGuard = new Object();
-  // Limit forced refreshes to 1 every minute.
-  private final RateLimiter<Object> forcedRenewRateLimiter =
-      RateLimiter.burstyBuilder(2, Duration.ofSeconds(30)).build();
-
+  // Rate limit delay calculator
```

Comment:
New rate limit calculator so we perform rate limit delays without blocking executor threads.